resteasy-client-microprofile/src/test/java/org/jboss/resteasy/microprofile/client/ProviderFactoryTest.java

```diff
@@ -0,0 +1,35 @@
+package org.jboss.resteasy.microprofile.client;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.jboss.resteasy.core.providerfactory.ResteasyProviderFactoryImpl;
+import org.jboss.resteasy.plugins.providers.DefaultTextPlain;
+import org.jboss.resteasy.plugins.providers.IIOImageProvider;
+import org.jboss.resteasy.spi.ResteasyProviderFactory;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ProviderFactoryTest {
+
+    @Test
+    public void testDefaultProvider() {
+        // Reset to ensure we start from clean slate
+        RestClientBuilderImpl.setProviderFactory(null);
+
+        RestClientBuilderImpl builder = (RestClientBuilderImpl) RestClientBuilder.newBuilder();
+
+        Assert.assertTrue(builder.getBuilderDelegate().getProviderFactory().getProviderClasses().contains(IIOImageProvider.class));
+        Assert.assertTrue(builder.getBuilderDelegate().getProviderFactory().getProviderClasses().contains(DefaultTextPlain.class));
+    }
+
+    @Test
+    public void testCustomProvider() {
+        ResteasyProviderFactory provider = new ResteasyProviderFactoryImpl(null, true);
+        provider.registerProvider(IIOImageProvider.class);
+        RestClientBuilderImpl.setProviderFactory(provider);
```

Comment:
This actually has a side-effect of leaving the RestClientBuilderImpl at the end of the test in a state that's different from the state it had before the test. This could lead to strange/unexpected failures in case of testsuite execution order changes, etc. Can you restore the original state using a try-finally block? Thanks.